lib/index.d.ts

```diff
@@ -0,0 +1,801 @@
+import type { Response } from "node-fetch"
+export { Response }
+
+export interface DinkyRequestOptions {
+  /**
+   * User's personal API key taken from their account.
+   */
+  key?: string
+
+  /**
+   * An ID of a filter found on filters page.
+   * The ID can be found on filters page: <https://derpibooru.org/filters>
+   */
+  filter?: number
+}
+
+export class Dinky {
+  /**
+   * Creates an instance of derpibooru API client.
+   * See `DinkyRequestOptions` for more details on configuration.
+   *
+   * Default options: `{}`
+   */
+  constructor(options?: DinkyRequestOptions)
+
+  /**
+   * Creates a request handler for `/api/v1/json/search`
+   *
+   * @param query – query params
+   */
+  search(...query: (string | string[])[]): ImagesSearch
+
+  /**
+   * Creates a request handler for `/api/v1/json/comments`
+   */
+  comments(): Comments
+
+  /**
+   * Creates a request handler for `/api/v1/json/tags`
+   */
+  tags(): Tags
+
+  /**
+   * Creates a request handler for `/api/v1/json/images`
+   */
+  images(): Images
+
+  /**
+   * Creates a `Lists` class that contains a bunch of shortcuts for `Search`
+   */
+  lists(): Lists
+}
+
+export function dinky(options: DinkyRequestOptions): Dinky
+export default dinky
+
+export class NetworkError extends Error {
+  response: Response
+  url: string
+  status: number
+  statusText: string
+
+  // TODO: should this one be private?
+  constructor(message: string, response: Response)
+}
+
+interface Request<T> {
+  /**
+   * Sets the page offset.
+   *
+   * Default offset: `1`
+   */
+  page(offset: number): this
+
+  /**
+   * Executes current request.
+   *
+   * Default options: `{}`
+   */
+  exec<T>(options?: DinkyRequestOptions): Promise<T>
+
+  then: Promise<T>["then"]
+  catch: Promise<T>["catch"]
+}
+
+// TODO: its constructor is marked as public, so why does it have @private annotation then?
+// I guess this doesn't mean it is public to external users, but public inside of this lib or smth...
```

Comment:
The only public class for external users is Dinky, anything else has `@api private` annotation. This is due to the library's design.

Reply:
Removed this comment in https://github.com/octet-stream/dinky/pull/7/commits/14cfd777a52f1ef319befeb18af6ba8eff751b26
`Entities` interface is left unexported, thus it is a private implementation detail.